src/map/clif.c

```diff
@@ -12454,7 +12454,7 @@ static void clif_parse_TradeRequest(int fd, struct map_session_data *sd)
 
 	struct map_session_data *t_sd = map->id2sd(RFIFOL(fd, 2));
 
-	if (sd->chat_id == 0 && pc_cant_act(sd))
+	if (pc_cant_act_except_npc_chat(sd) || (sd->npc_id != 0 && sd->state.using_megaphone == 0))
```

Comment:
look like existing check here was wrong.
probably here need use ``pc_cant_act_except_npc``?

Reply:
Nope, that's correct. I tested on Aegis and you indeed are allowed to trade while being inside a chat room.